core/src/main/scala/chisel3/Module.scala

```diff
@@ -376,7 +376,13 @@ package experimental {
     private val _ports = new ArrayBuffer[(Data, SourceInfo)]()
 
     // getPorts unfortunately already used for tester compatibility
-    protected[chisel3] def getModulePorts = {
+    protected[chisel3] def getModulePorts: Seq[Data] = {
+      require(_closed, "Can't get ports before module close")
+      _ports.map(_._1).toSeq
```

Comment:
```suggestion
      _ports.iterator.map(_._1).toSeq
```
Little micro-optimization but saves a List copy.